airflow/providers/sas/_utils/logon.py

```diff
@@ -0,0 +1,95 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import base64
+import urllib.parse
+
+import requests
+import urllib3
+from urllib3.exceptions import InsecureRequestWarning
+
+from airflow.hooks.base import BaseHook
+
+
+def create_session_for_connection(connection_name: str):
+    print(f"Creating session for connection named {connection_name}")
```

Comment:
We use Hooks for this purpose: https://airflow.apache.org/docs/apache-airflow/stable/concepts/connections.html#hooks which is also provide nice ability to create Connection into the UI: https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html#custom-connection-types